### PR TITLE
feat: makes containers port(s) list multi line

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

To be able to have a `Forward` button on the container's ports, we need to have a little bit more space. This PR change the single line all ports in one, to a multi line per port.

See https://github.com/containers/podman-desktop/issues/9273#issuecomment-2432358681 for the mockup I made (reason why we need more space)

> :warning: this code cannot be reused for service as `container.ports` are `Array<V1ContainerPort>` and `services.port` are `V1ServicePort` and they do not share the same ports.

### Screenshot / video of UI

**Before**

image

**After**

image

### What issues does this PR fix or reference?

Required for https://github.com/containers/podman-desktop/issues/9273

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Comment by benoitf** [comment]
suggestion: rather than saying 'we need more space', explain how the feature/button will appear. Then people can say if it'll help or not for the mockup.

**Comment by axel7083** [comment]
> suggestion: rather than saying 'we need more space', explain how the feature/button will appear. Then people can say if it'll help or not for the mockup.

Yes! Sorry I wanted to put a comment on the issue and I forgot: updated now :+1:

File: packages/renderer/src/lib/kube/details/KubeContainerPorts.svelte [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import Cell from '/@/lib/details/DetailsCell.svelte';
+
+interface Props {
+  ports?: V1ContainerPort[];
+}
+
+let { ports }: Props = $props();
+</script>
+
+{#if ports && ports.length > 0}
```
Comment (deboer-tim) [inline packages/renderer/src/lib/kube/details/KubeContainerPorts.svelte:13]: `ports?.length > 0`?

File: packages/renderer/src/lib/kube/details/KubeContainerPorts.spec.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        containerPort: 80,
+      },
+    ],
+  });
+
+  const title = getByText('Ports');
+  expect(title).toBeDefined();
+});
+
+test('expect multiple ports to be visible', async () => {
```
Comment (deboer-tim) [inline packages/renderer/src/lib/kube/details/KubeContainerPorts.spec.ts:46]: If the point of the PR was to switch them to a column, maybe we should test parent to confirm flex-col?

**Review by deboer-tim — APPROVED** [review]
Looks good, easier to read.

**Comment by benoitf** [comment]
> Yes! Sorry I wanted to put a comment on the issue and I forgot: updated now 👍

suggestion: add the button in the image where it will added so people can quickly see why is it needed, how this pr will solve the problem and then assert that it's the best path

File: packages/renderer/src/lib/details/DetailsCell.svelte [hunk 3]
```diff
@@ -3,6 +3,6 @@ export let style: string = '';
 export let onClick: () => void = () => {};
 </script>
 
-<td class="pt-1 pl-3 {style}" on:click={onClick}>
+<td class="pt-1 pl-3 {style} {$$props.class}" on:click={onClick}>
```
Comment (benoitf) [inline packages/renderer/src/lib/details/DetailsCell.svelte:6]: suggestion (non-blocking): move to runes

**Review by benoitf — APPROVED** [review]

**Review by jeffmaury — APPROVED** [review]
LGTM

File: packages/renderer/src/lib/kube/details/KubeContainerPorts.svelte (hunk 1 above)
Comment (axel7083) [reply packages/renderer/src/lib/kube/details/KubeContainerPorts.svelte:13]: Error: 'ports.length' is possibly 'undefined'. (ts)

{#if ports?.length > 0}

:(

**Pull request merged**